### ISSUE TITLE
Load data repeatedly after start training ?

### ISSUE THREAD

**Issue by JohnnyY8** [rbgirshick/py-faster-rcnn#252]
Hi, everyone:

I started training faster r-cnn on ImageNet VID training set, but some strange problem occurred. I printed some path in train.txt, like this:
![image](https://cloud.githubusercontent.com/assets/12611573/16788275/242bc476-48d6-11e6-8359-e13171006bc5.png)

When I see the printed information in terminal, I notice that the data has been loaded for many times! My teammate and I are pretty sure it has finished the whole training set for at least once. But this information shows it start from 0000 again.
![image](https://cloud.githubusercontent.com/assets/12611573/16788282/3a460866-48d6-11e6-9005-979cf9ab814a.png)

Can anyone help me? We have loaded training data for more than 40 hours.
Thank you so much!

**Comment by manipopopo** [comment]
You may try to search `image_path_from_index` and `image_path_at` through the source code.

You'll find that `prepare_roidb` in [roidb.py](https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/roi_data_layer/roidb.py#L16) calls `image_path_at` twice. When `cfg.TRAIN.USE_FLIPPED` is `True`, py-faster-rcnn [calls](https://github.com/rbgirshick/py-faster-rcnn/blob/master/lib/datasets/imdb.py#L102`s) `image_path_at` and loads all images to get widths.

So (in the original implementation) images are loaded to memory twice. `image_path_from_index(some_index)` is called three times.

By the way, maybe you can skip `000141.JPEG`-`000169.JPEG` if they are quite similar to `ILSVRC2015_train_00034000/000140.JPEG`. The whole data set is somewhat too large.

**Comment by JohnnyY8** [comment]
@manipopopo Thanks for your advice.
So if I see print information for three times, it is actually ok, right ?
I will try now.

**Comment by manipopopo** [comment]
If you're really worried about the correctness of your code, maybe you can start with a training on a `train.txt` containing only two records.

**Comment by JohnnyY8** [comment]
@manipopopo Yeah, I know what you mean, I will try small dataset to start training to check if my code is right.  Thank you so much！
